utils/plugins.js

```diff
@@ -0,0 +1,30 @@
+/* eslint no-console: [ 'error', { allow: [ 'error' ] } ] */
+
+/**
+ * Validates the namespaced id.
+ *
+ * A namespaced id should be formatted like this: "namespace/identifier".
+ * It can only contain lowercase alphanumeric characters and dashes.
+ *
+ * If a namespace id is invalid, it will print a console error, optionally using
+ * the name of the thing it identifies.
+ *
+ * @param {string} id Namespaced identifier.
+ * @param {string} [name] Plural name of what the id is for, defaults to "Namespaced identifiers".
+ * @return {boolean} Whether the id is valid.
+ */
+export function validateNamespacedId( id, name = 'Namespaced identifiers' ) {
+	if ( typeof id !== 'string' ) {
+		console.error(
+			wp.i18n.sprintf( '%s must be strings.', name )
```

Comment:
Sorry, I know I recommended this, but import semantics are preferred in place of these globals:

```js
import { sprintf } from '@wordpress/i18n';
```

Related discussion: https://wordpress.slack.com/archives/C02QB2JS7/p1519839851000186

Reply:
FWIW, while it probably has little impact here in that this warning is very infrequently logged, but I expect `sprintf` being not a language-native function is considerably less performant than a simple string concatenation.